packages/core-utils/src/base-service.ts

```diff
@@ -24,6 +26,7 @@ export class BaseService<T> {
     this.name = name
     this.options = mergeDefaultOptions(options, optionSettings)
     this.logger = new Logger({ name })
+    this.metrics = new Metrics({ prefix: name.split(' ').join('_') })
```

Comment:
Hm, this may be a bit annoying. Does the metrics client dislike spaces in service names? Perhaps we should refactor the `BaseService` such that names cannot have spaces in them so that the logs and metrics use identical names.

Reply:
https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels only underscores and colons 😬 honestly might be the move for cleanliness sake - thoughts on importing this validation and enforcing it? https://github.com/siimon/prom-client/blob/96f7495d66b1a21755f745b1367d3e530668a957/lib/validation.js#L5-L11 or should our validation be looser?

Reply:
just renamed everything without adding the validation since we will get the error on metrics initialization